include/aws/common/json.h

```diff
@@ -0,0 +1,395 @@
+#ifndef AWS_COMMON_JSON_H
+#define AWS_COMMON_JSON_H
+
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include "thread.h"
+
+struct aws_json_node;
+
+// ====================
+// Create and pass type
+
+/**
+ * Creates a new aws_json_node and returns a pointer to it.
+ *
+ * Note: You will need to free the memory for the aws_json_node using aws_json_delete on the aws_json_node or
+ * on the object/array containing the aws_json_node.
+ * @return A new aws_json_node
+ */
+AWS_COMMON_API
+struct aws_json_node *aws_json_node_new(void);
+
+/**
+ * Creates a new string aws_json_node with the given string and returns a pointer to it.
+ *
+ * Note: You will need to free the memory for the aws_json_node using aws_json_delete on the aws_json_node or
+ * on the object/array containing the aws_json_node.
+ * @param string The string you want to store in the aws_json_node
+ * @return A new string aws_json_node
+ */
+AWS_COMMON_API
+struct aws_json_node *aws_json_string_new(char *string);
```

Comment:
This may be a bit painful (especially for the input parameters), but there shouldn't be any C-strings in the public API, either as parameters or return values.  Use byte cursors instead and for parameters that are passed directly to CJSON, you'll need to create an aws_string, pass in the bytes, and then destroy the string.